cli/lsp/diagnostics.rs

```diff
@@ -1376,4 +1393,64 @@ let c: number = "a";
       })
     );
   }
+
+  #[tokio::test]
+  async fn duplicate_diagnostics_for_duplicate_imports() {
+    let temp_dir = TempDir::new();
+    let (snapshot, _) = setup(
+      &temp_dir,
+      &[(
+        "file:///a.ts",
+        r#"
+        import "bad.js";
+        import "bad.js";
```

Comment:
Maybe try adding a test with multiple bad `@deno-types` directives as well?

Reply:
Only one `@deno-types` directive is stored per JS dep, that would be addressed by https://github.com/denoland/deno_graph/pull/272.